src/components/action-bar/action-bar.tsx

```diff
@@ -170,10 +189,41 @@ export class ActionBar {
                 items={items}
                 onSelect={this.handleSelect}
                 role="gridcell"
+                actionBarIsShrunk={this.actionBarIsShrunk}
             />
         );
     };
 
+    private renderCollapseExpandButton() {
+        if (!this.isExpandable) {
+            return;
+        }
+
+        return (
+            <button
+                class={{
+                    'expand-shrink': true,
+                }}
+                aria-label={this.tooltipLabel}
+                type="button"
+                onClick={this.handleCollapseExpandClick}
+            >
+                <limel-icon
+                    name="double_left"
+                    id="tooltip-expand-shrink-button"
+                />
+                <limel-tooltip
+                    label={this.tooltipLabel}
+                    elementId="tooltip-expand-shrink-button"
+                />
+            </button>
+        );
+    }
+
+    private handleCollapseExpandClick = () => {
+        this.actionBarIsShrunk = !this.actionBarIsShrunk;
```

Comment:
You can't mutate a prop! :)

Reply:
Yes, we have done it in other places. That doesn't mean it's a good practice. We should avoid it, unless we have very good reason to use it.

Reply:
I know @adrianschmidt I am just spreading the knowledges that it is possible to mutate

Reply:
Yes I know it's possible but that decorators extra parameter "mutable" should really be used with caution.